packages/tupaia-web-server/src/routes/DashboardsRoute.ts

```diff
@@ -82,77 +70,78 @@ export class DashboardsRoute extends Route<DashboardsRequest> {
     const { projectCode, entityCode } = params;
 
     // We're including the root entity in this request, so we don't need to double up fetching it
-    const entities = await ctx.services.entity.getAncestorsOfEntity(
+    const entities: Entity[] = await ctx.services.entity.getAncestorsOfEntity(
       projectCode,
       entityCode,
       {},
       true,
     );
-    const rootEntity = entities.find((e: Entity) => e.code === entityCode);
+    const rootEntity = ensure(entities.find(e => e.code === entityCode));
 
     const rootEntityPermissions = rootEntity.country_code
       ? accessPolicy.getPermissionGroups([rootEntity.country_code])
       : accessPolicy.getPermissionGroups(); // country_code is null for project level
 
-    const dashboards: Dashboard[] = await this.req.models.dashboard.find(
+    const dashboards = await this.req.models.dashboard.find(
       {
-        root_entity_code: entities.map((e: Entity) => e.code),
+        root_entity_code: entities.map(e => e.code),
```

Comment:
There's a lot of explicitly typing function arguments in our code which doesn't need to be there (TS will infer the type based of the array type). I have a strong preference for letting typescript do inference where possible as it keeps the code cleaner and doesn't cause so many issues when we alter the type definitions.